zenoh-jni/src/query.rs

```diff
@@ -66,19 +70,27 @@ pub(crate) unsafe extern "C" fn Java_io_zenoh_jni_JNIQuery_replySuccessViaJNI(
     _class: JClass,
     query_ptr: *const zenoh::queryable::Query,
     key_expr_ptr: *const KeyExpr<'static>,
+    key_expr_str: JString,
     payload: JByteArray,
     encoding: jint,
     sample_kind: jint,
     timestamp_enabled: jboolean,
     timestamp_ntp_64: jlong,
     attachment: JByteArray,
 ) {
-    let key_expr = Arc::from_raw(key_expr_ptr);
-    let key_expr_clone = key_expr.deref().clone();
-    std::mem::forget(key_expr);
+    let key_expr: KeyExpr<'static> = match process_key_expr(&mut env, &key_expr_str, key_expr_ptr) {
+        Ok(key_expr) => key_expr,
+        Err(err) => {
+            _ = err
+                .throw_on_jvm(&mut env)
+                .map_err(|err| tracing::error!("{}", err));
+            return;
+        }
+    };
+
```

Comment:
Look into  `map_err` When ever encountering the pattern 
```rust 
match x {
    Ok(y) => y,
    Err(err) => {
    // handle error
    }
} 
```

Something like 

```rust 
x.map_err(|err| { 
    // handle error
   })
```

Reply:
In this case I need to be able to stop the function in case of error (we throw an exception and stop the function), thus the match with the `return`.